Bug: test failures

### Description

The tests ran right around midnight UTC time.
```
________________ test_default_serializer[v1-condate-2024-05-08] ________________
[gw0] linux -- Python 3.8.18 /home/runner/work/litestar/litestar/.venv/bin/python

model = ModelV1(custom_str='', custom_int=0, custom_float=0.0, custom_list=[], custom_set=CustomSet(), custom_frozenset=Custom...scheme='some', host='example.org', tld='org', host_type='domain', path='/'), http_url=HttpUrl('http://example.org/', ))
attribute_name = 'condate', expected = '2024-05-08'

    @pytest.mark.parametrize(
        "attribute_name, expected",
        [
            ("path", "example"),
            ("email_str", "info@example.org"),
            ("name_email", "info <info@example.org>"),
            ("color", "white"),
            ("bytesize", 100),
            ("secret_str", "**********"),
            ("secret_bytes", "**********"),
            ("payment_card_number", "4000000000000002"),
            ("constr", "hello"),
            ("conbytes", b"hello"),
            ("condate", datetime.date.today().isoformat()),
            ("condecimal", 3.14),
            ("conset", {1}),
            ("confrozenset", frozenset([1])),
            ("conint", 1),
            ("url", "some://example.org/"),
            ("http_url", "http://example.org/"),
        ],
    )
    def test_default_serializer(model: ModelV1 | ModelV2, attribute_name: str, expected: Any) -> None:
>       assert serializer(getattr(model, attribute_name)) == expected
E       AssertionError: assert '2024-05-09' == '2024-05-08'
E         - 2024-05-08
E         ?          ^
E         + 2024-05-09
E         ?          ^

tests/unit/test_contrib/test_pydantic/test_plugin_serialization.py:203: AssertionError
________________ test_default_serializer[v2-condate-2024-05-08] ________________
[gw1] linux -- Python 3.8.18 /home/runner/work/litestar/litestar/.venv/bin/python

model = ModelV2(conset={1}, confrozenset=frozenset({1}), conlist=[1], path=PosixPath('example'), email_str='info@example.org',...ondecimal=Decimal('3.14'), confloat=1.0, conint=1, url=Url('some://example.org/'), http_url=Url('http://example.org/'))
attribute_name = 'condate', expected = '2024-05-08'

    @pytest.mark.parametrize(
        "attribute_name, expected",
        [
            ("path", "example"),
            ("email_str", "info@example.org"),
            ("name_email", "info <info@example.org>"),
            ("color", "white"),
            ("bytesize", 100),
            ("secret_str", "**********"),
            ("secret_bytes", "**********"),
            ("payment_card_number", "4000000000000002"),
            ("constr", "hello"),
            ("conbytes", b"hello"),
            ("condate", datetime.date.today().isoformat()),
            ("condecimal", 3.14),
            ("conset", {1}),
            ("confrozenset", frozenset([1])),
            ("conint", 1),
            ("url", "some://example.org/"),
            ("http_url", "http://example.org/"),
        ],
    )
    def test_default_serializer(model: ModelV1 | ModelV2, attribute_name: str, expected: Any) -> None:
>       assert serializer(getattr(model, attribute_name)) == expected
E       AssertionError: assert '2024-05-09' == '2024-05-08'
E         - 2024-05-08
E         ?          ^
E         + 2024-05-09
E         ?          ^

tests/unit/test_contrib/test_pydantic/test_plugin_serialization.py:203: AssertionError
```

and 

```
_______________ test_default_serializer[v1-condate-2024-05-08] ________________
[gw0] win32 -- Python 3.12.3 D:\a\litestar\litestar\.venv\Scripts\python.exe

model = ModelV1(custom_str='', custom_int=0, custom_float=0.0, custom_list=[], custom_set=CustomSet(), custom_frozenset=Custom...scheme='some', host='example.org', tld='org', host_type='domain', path='/'), http_url=HttpUrl('http://example.org/', ))
attribute_name = 'condate', expected = '2024-05-08'

    @pytest.mark.parametrize(
        "attribute_name, expected",
        [
            ("path", "example"),
            ("email_str", "info@example.org"),
            ("name_email", "info <info@example.org>"),
            ("color", "white"),
            ("bytesize", 100),
            ("secret_str", "**********"),
            ("secret_bytes", "**********"),
            ("payment_card_number", "4000000000000002"),
            ("constr", "hello"),
            ("conbytes", b"hello"),
            ("condate", datetime.date.today().isoformat()),
            ("condecimal", 3.14),
            ("conset", {1}),
            ("confrozenset", frozenset([1])),
            ("conint", 1),
            ("url", "some://example.org/"),
            ("http_url", "http://example.org/"),
        ],
    )
    def test_default_serializer(model: ModelV1 | ModelV2, attribute_name: str, expected: Any) -> None:
>       assert serializer(getattr(model, attribute_name)) == expected
E       AssertionError: assert '2024-05-09' == '2024-05-08'
E         - 2024-05-08
E         ?          ^
E         + 2024-05-09
E         ?          ^

tests\unit\test_contrib\test_pydantic\test_plugin_serialization.py:203: AssertionError
_______________ test_default_serializer[v2-condate-2024-05-08] ________________
[gw0] win32 -- Python 3.12.3 D:\a\litestar\litestar\.venv\Scripts\python.exe

model = ModelV2(conset={1}, confrozenset=frozenset({1}), conlist=[1], path=WindowsPath('example'), email_str='info@example.org...ondecimal=Decimal('3.14'), confloat=1.0, conint=1, url=Url('some://example.org/'), http_url=Url('http://example.org/'))
attribute_name = 'condate', expected = '2024-05-08'

    @pytest.mark.parametrize(
        "attribute_name, expected",
        [
            ("path", "example"),
            ("email_str", "info@example.org"),
            ("name_email", "info <info@example.org>"),
            ("color", "white"),
            ("bytesize", 100),
            ("secret_str", "**********"),
            ("secret_bytes", "**********"),
            ("payment_card_number", "4000000000000002"),
            ("constr", "hello"),
            ("conbytes", b"hello"),
            ("condate", datetime.date.today().isoformat()),
            ("condecimal", 3.14),
            ("conset", {1}),
            ("confrozenset", frozenset([1])),
            ("conint", 1),
            ("url", "some://example.org/"),
            ("http_url", "http://example.org/"),
        ],
    )
    def test_default_serializer(model: ModelV1 | ModelV2, attribute_name: str, expected: Any) -> None:
>       assert serializer(getattr(model, attribute_name)) == expected
E       AssertionError: assert '2024-05-09' == '2024-05-08'
E         - 2024-05-08
E         ?          ^
E         + 2024-05-09
E         ?          ^

tests\unit\test_contrib\test_pydantic\test_plugin_serialization.py:203: AssertionError
============================== warnings summary ===============================
tests/unit/test_concurrency.py::test_sync_to_thread_trio
  D:\a\litestar\litestar\.venv\Lib\site-packages\trio\_core\_wakeup_socketpair.py:59: RuntimeWarning: It looks like Trio's signal handling code might have collided with another library you're using. If you're running Trio in guest mode, then this might mean you should set host_uses_signal_set_wakeup_fd=True. Otherwise, file a bug on Trio and we'll help you figure out what's going on.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ===========================
FAILED tests/unit/test_contrib/test_pydantic/test_plugin_serialization.py::test_default_serializer[v1-condate-2024-05-08] - AssertionError: assert '2024-05-09' == '2024-05-08'
  - 2024-05-08
  ?          ^
  + 2024-05-09
  ?          ^
FAILED tests/unit/test_contrib/test_pydantic/test_plugin_serialization.py::test_default_serializer[v2-condate-2024-05-08] - AssertionError: assert '2024-05-09' == '2024-05-08'
  - 2024-05-08
  ?          ^
  + 2024-05-09
  ?          ^
= 2 failed, 5043 passed, 169 skipped, 7 xfailed, 5 xpassed, 1 warning in 72.62s (0:01:12) =
```

https://github.com/litestar-org/litestar/actions/runs/9010091640/job/24755515232
https://github.com/litestar-org/litestar/actions/runs/9010091640/job/24755515608

### URL to code causing the issue

_No response_

### MCVE

```python
# Your MCVE code here
```


### Steps to reproduce

```bash
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error
```


### Screenshots

```bash
"![SCREENSHOT_DESCRIPTION](SCREENSHOT_LINK.png)"
```


### Logs

_No response_

### Litestar Version

main

### Platform

- [X] Linux
- [ ] Mac
- [X] Windows
- [ ] Other (Please specify in the description above)

<!-- POLAR PLEDGE BADGE START -->
---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/3481">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/3481/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/3481/pledge.svg">
</picture>
</a>
<!-- POLAR PLEDGE BADGE END -->

--- a/tests/unit/test_contrib/test_pydantic/test_plugin_serialization.py
+++ b/tests/unit/test_contrib/test_pydantic/test_plugin_serialization.py
@@ -27,6 +27,8 @@ from litestar.serialization import (
 
 from . import PydanticVersion
 
+TODAY = datetime.date.today()
+
 
 class CustomStr(str):
     pass
@@ -84,7 +86,7 @@ class ModelV1(pydantic_v1.BaseModel):
 
     constr: pydantic_v1.constr(min_length=1)  # type: ignore[valid-type]
     conbytes: pydantic_v1.conbytes(min_length=1)  # type: ignore[valid-type]
-    condate: pydantic_v1.condate(ge=datetime.date.today())  # type: ignore[valid-type]
+    condate: pydantic_v1.condate(ge=TODAY)  # type: ignore[valid-type]
     condecimal: pydantic_v1.condecimal(ge=Decimal("1"))  # type: ignore[valid-type]
     confloat: pydantic_v1.confloat(ge=0)  # type: ignore[valid-type]
 
@@ -112,7 +114,7 @@ class ModelV2(pydantic_v2.BaseModel):
 
     constr: pydantic_v2.constr(min_length=1)  # type: ignore[valid-type]
     conbytes: pydantic_v2.conbytes(min_length=1)  # type: ignore[valid-type]
-    condate: pydantic_v2.condate(ge=datetime.date.today())  # type: ignore[valid-type]
+    condate: pydantic_v2.condate(ge=TODAY)  # type: ignore[valid-type]
     condecimal: pydantic_v2.condecimal(ge=Decimal("1"))  # type: ignore[valid-type]
     confloat: pydantic_v2.confloat(ge=0)  # type: ignore[valid-type]
 
@@ -144,7 +146,7 @@ def model(pydantic_version: PydanticVersion) -> ModelV1 | ModelV2:
             payment_card_number=pydantic_v1.PaymentCardNumber("4000000000000002"),
             constr="hello",
             conbytes=b"hello",
-            condate=datetime.date.today(),
+            condate=TODAY,
             condecimal=Decimal("3.14"),
             confloat=1.0,
             conset={1},
@@ -165,7 +167,7 @@ def model(pydantic_version: PydanticVersion) -> ModelV1 | ModelV2:
         payment_card_number=pydantic_v2.PaymentCardNumber("4000000000000002"),
         constr="hello",
         conbytes=b"hello",
-        condate=datetime.date.today(),
+        condate=TODAY,
         condecimal=Decimal("3.14"),
         confloat=1.0,
         conset={1},
@@ -190,7 +192,7 @@ def model(pydantic_version: PydanticVersion) -> ModelV1 | ModelV2:
         ("payment_card_number", "4000000000000002"),
         ("constr", "hello"),
         ("conbytes", b"hello"),
-        ("condate", datetime.date.today().isoformat()),
+        ("condate", TODAY.isoformat()),
         ("condecimal", 3.14),
         ("conset", {1}),
         ("confrozenset", frozenset([1])),

